src/lib/mina_base/parties_logic.ml

```diff
@@ -444,9 +449,17 @@ module Make (Inputs : Inputs_intf) = struct
             Transaction_commitment.if_ is_start' ~then_:on_start
               ~else_:local_state.transaction_commitment
       in
+      let full_commitment =
+        match is_start with
+        | `No ->
+            local_state.full_commitment
+        | `Yes _ | `Compute _ ->
+            transaction_commitment
```

Comment:
I think you'll want to add a call here (like the above `Transaction_commitment_on_start`) that performs the fee-payer inclusion. Then, that can return [this value](https://github.com/MinaProtocol/mina/blob/dd2c5270ebfe0d2fcc92d312cc27261015c2f496/src/lib/transaction_snark/transaction_snark.ml#L2002), and the call to `Check_auth_and_update_account` can take the full commitment and reuse it.